mLRS/Common/stack_check.h

```diff
@@ -0,0 +1,52 @@
+//*******************************************************
+// Copyright (c) MLRS project
+// GPL3
+// https://www.gnu.org/licenses/gpl-3.0.de.html
+// OlliW @ www.olliw.eu
+//*******************************************************
+// Stack check
+//*******************************************************
+#ifndef STACK_CHECK_H
+#define STACK_CHECK_H
+#pragma once
+
+
+#include <inttypes.h>
+
+
+//-------------------------------------------------------
+//
+//-------------------------------------------------------
+// see
+// - https://mcuoneclipse.com/2016/11/01/getting-the-memory-range-of-sections-with-gnu-linker-files/
+// - https://stackoverflow.com/questions/55622174/is-accessing-the-value-of-a-linker-script-variable-undefined-behavior-in-c
+extern uint32_t _estack;
+extern uint32_t _Min_Stack_Size;
+
+
+uint32_t stack_check_used(void)
+{
+    uint32_t ptr = (uint32_t)&_estack - (uint32_t)&_Min_Stack_Size;
+    uint32_t end = (uint32_t)&_estack;
+    while (ptr < end) {
+        if (*(uint8_t*)ptr != 0xAA) break;
+        ptr++;
+    }
+
+    return (end - ptr);
+}
+
+
+void stack_check_init(void)
+{
+    // paint the stack memory area
+    uint32_t ptr = (uint32_t)&_estack - (uint32_t)&_Min_Stack_Size;
+    uint32_t end = (uint32_t)&_estack;
+    while (ptr < end) {
+        *(uint8_t*)ptr = 0xAA;
+        ptr++;
```

Comment:
Isn't the pointer incrementing here with 4 bytes/32-bits. Also line 33 above?
I would suggest to use:
```c++
*ptr = 0xAA55AA55;
ptr++;
```

Reply:
was worried first about this too, but IMHO not, since ptr is not a pointer but just a uint32_t variable

this refutes also your suggestion :)

Reply:
Would need to verify with in-circuit-debugger what happens with ptr++.
Alternatively:
```c++
if (*(uint32_t*)ptr != 0xAAAAAAAA) break;
```
?

Reply:
it is a ptr in that you access the loaction with it, i.e. use *ptr to set it. But pointer are just uint32_t by itself, since they are just 32bit adressess. So, I think it's all fine. I sure admit though that I also puzzeled few moments what exactkly needs to be done here.